### ISSUE TITLE
[a11y]: Tearsheet: enable (or simplify) proper use of heading ranks in content

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#7164]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.62

### React version

18

### Automated testing tool and ruleset

https://www.w3.org/TR/WAI-WEBCONTENT/wai-pageauth.html#tech-logical-headings

### Assistive technology

https://www.w3.org/WAI/tutorials/page-structure/headings/

### Description

According to https://medium.com/@web-accessibility-education/which-heading-level-should-dialog-modals-have-7b3df89437f0,
a dialog’s top heading should be either H1 or H2.   By corollary, the heading inside the dialog needs to start one level lower,
at H2 or H3.
    
However, Tearsheet has a strange design w.r.t. heading levels
    
- Sometimes has both H2 and H3 headings, for example at
    https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-tearsheet--with-navigation&globals=viewport:basic,
- Sometimes just has an H3 heading (for example at
    https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-tearsheet--tearsheet&globals=viewport:basic).
    
I’m skeptical if that makes sense, especially in the second case, but this ticket is just about making sure the headings inside the Tearsheet mesh with the Tearsheet's own H3 heading.  In other words, the headings inside the Tearsheet should start at H4, even if the content is written (as encouraged by Carbon) using `Section` and `Heading`.

### WCAG 2.1 Violation

https://www.w3.org/TR/WAI-WEBCONTENT/wai-pageauth.html#tech-logical-headings

### Reproduction/example

https://stackblitz.com/edit/github-ykbrpwcg-8dssjoc4?file=src%2FExample%2FExample.jsx

### Steps to reproduce

Just check heading level's in Chrome's developer tools.   Alternately you could use the WAVE plugin.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
